src/viam/sdk/components/base/base.cpp

```diff
@@ -34,7 +34,7 @@ std::shared_ptr<ResourceRegistration> Base::resource_registration() {
     const google::protobuf::ServiceDescriptor* sd =
         p->FindServiceByName(viam::component::base::v1::BaseService::service_full_name());
     if (!sd) {
-        throw std::runtime_error("Unable to get service descriptor for the base service");
+        throw ViamException("Unable to get service descriptor for the base service");
```

Comment:
just a heads up that all of these "Unable to get service descriptor for {resourcetype} service" errors in `{resourcetype}.cpp` are being removed in the `refactor-registry` PR so some rebase conflict will be happening!